selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java

```diff
@@ -213,22 +219,55 @@ public void updateFile(String workspaceId, String pathToFile, String content) th
     }
   }
 
-  private String getWsAgentUrl(String workspaceId) throws Exception {
-    Workspace workspace = workspaceServiceClient.getById(workspaceId);
-    workspaceServiceClient.ensureRunningStatus(workspace);
-
-    Map<String, ? extends Machine> machines =
-        workspaceServiceClient.getById(workspaceId).getRuntime().getMachines();
-    for (Machine machine : machines.values()) {
-      if (containsWsAgentServer(machine)) {
-        Server wsAgentServer = machine.getServers().get(SERVER_WS_AGENT_HTTP_REFERENCE);
-        if (wsAgentServer != null) {
-          return wsAgentServer.getUrl() + "/project";
-        } else {
-          throw new RuntimeException("Workspace agent server is null");
-        }
-      }
-    }
-    throw new RuntimeException("Cannot find dev machine on workspace");
+  public boolean checkProjectType(String wokspaceId, String projectName, String projectType)
+      throws Exception {
+    return getProjectParameters(wokspaceId, projectName)
+        .get("type")
+        .getAsString()
+        .equals(projectType);
+  }
+
+  public boolean checkProjectLanguage(String wokspaceId, String projectName, String projectLanguage)
+      throws Exception {
+    return getProjectParameters(wokspaceId, projectName)
+        .getAsJsonObject("attributes")
+        .get("language")
+        .getAsString()
+        .equals(projectLanguage);
+  }
+
+  public List<String> getExternalLibraries(String workspaceId, String projectName)
+      throws Exception {
+    List<String> result = new ArrayList<>();
+    HttpJsonResponse response =
+        requestFactory
+            .fromUrl(workspaceAgentApiEndpoint.get(workspaceId) + "java/navigation/libraries")
+            .useGetMethod()
+            .addQueryParam("projectpath", "/" + projectName)
+            .request();
+
```

Comment:
I think using DTO will be more reliable - this will be able to avoid potential problems after changing 'Jar (library) object'.  Also it will make code shorter. 
**For example add dependency:**
```xml 
     <dependency>
            <groupId>org.eclipse.che.plugin</groupId>
            <artifactId>che-plugin-java-parent</artifactId>
        </dependency>
```
**And try to use code like:**
```java
result= DtoFactory.getInstance().createListDtoFromJson(response.asString(), Jar.class).stream().map(e->e.getName()).collect( Collectors.toList());
```
But this is only my opinion.